sdmetrics/single_table/privacy/dcr_baseline_protection.py

```diff
@@ -0,0 +1,199 @@
+"""DCR Baseline Protection metrics."""
+
+from sdmetrics.goal import Goal
+from sdmetrics.single_table.base import SingleTableMetric
+from sdmetrics.single_table.privacy.dcr_utils import calculate_dcr
+from sdmetrics.utils import get_columns_from_metadata, get_type_from_column_meta
+
+
+class DCRBaselineProtection(SingleTableMetric):
+    """DCR Baseline Protection metric."""
+
+    name = 'DCRBaselineProtection'
+    goal = Goal.IGNORE
+    min_value = 0.0
+    max_value = 1.0
+
+    @classmethod
+    def _validate_inputs(
+        cls,
+        real_training_data,
+        synthetic_data,
+        real_validation_data,
+        metadata,
+        num_rows_subsample,
+        num_iterations,
+    ):
+        if num_rows_subsample is not None:
+            if not isinstance(num_rows_subsample, int):
+                raise ValueError('num_rows_subsample must be an integer.')
+            if num_rows_subsample < 1:
+                raise ValueError(
+                    f'num_rows_subsample ({num_rows_subsample}) must be greater than 1.'
+                )
+        elif num_rows_subsample is None and num_iterations > 1:
+            raise ValueError(
+                'num_iterations should not be greater than 1 if there is not subsampling.'
+            )
+
+        if not isinstance(num_iterations, int):
+            raise ValueError('num_iterations must be an integer.')
+
+        if num_iterations < 1:
+            raise ValueError(f'num_iterations ({num_iterations}) must be greater than 1.')
+
+        if metadata is not None:
+            if not isinstance(metadata, dict):
+                metadata = metadata.to_dict()
+
+        valid_sdtypes = {'numerical', 'categorical', 'boolean', 'datetime'}
```

Comment:
@R-Palazzo I think you define this again in one of your PRs. Can we move this list to a centralized location?